### PR TITLE
Feature: Ability to find a route by a path

### PR DESCRIPTION
**Pull request by jquatier**
I have a use case where I'm adding routes to the server dynamically after the server is up. I would like to be able to check if a route exists for a path before I add it, because if another route already has a similar path regex, the new route will not fire (the first route that matches the request is always the one returned from the router).

Example:

```javascript
// existing route
server.get('/users/:id', () => {});   // regex: /^\/+users\/+([^\/]*)[\/]*$/
// new route
server.get('/users/:myParamName', () => {});   // regex: /^\/+users\/+([^\/]*)[\/]*$/
```

In this case, I wouldn't want to add the new route because the existing route has the same regex even though the parameter names are different. I would like to throw an error that says a route at that path already exists on the server, and then output the details of that route.

I didn't see another way to do this since the `compileUrl` method in router.js method isn't exported and that's what translates a path with parameter names to a RegEx. So rather than copy that method, I created this new method to translate a path to a RegEx and compare it against the route collection. I made it a `findByPath` method so that it can return the route that had the same path.

The other way I thought about doing it was to do my own parsing of the path and then set that as the route name (the `mount` method in the router will return `false` for routes that have the same name). But, I thought it might be good to still be able to name my routes differently from the path.

Anyway, TLDR; let me know if you have any feedback! Thanks! 👍

### CODE REVIEW

**Comment by micahr** [comment]
Trying to kick travis into gear, going to close and reopen to see if that helps.

**Comment by micahr** [comment]
PR looks good to me, thanks for adding the helper.

**Comment by micahr** [comment]
@yunong or @DonutEspresso any concerns?

**Comment by yunong** [comment]
If we can add some JSdocs then it looks good to merge.

**Comment by DonutEspresso** [comment]
Agreed - a blurb in the docs with an example usage would be great. LGTM!

**Comment by micahr** [comment]
Looks good with the JSDoc addition. I'd settle for filing an issue for adding docs to the new doc site once it becomes available.
